Ska/engarchive/time_adjust.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Describe what this module does.
+"""
+
+import copy
+
+from Chandra.Time import DateTime
+from Ska.engarchive.utils import logical_intervals
+from Ska.engarchive import fetch_eng as fetch
+from Ska.tdb import tables
+
+
+def adjust_time(msid, ts, tp):
+    """
+    Describe what the function does in one sentence.
+
+    Describe what the function does in more detail. Include notes about usage,
+    for instance the important caveat about the MSID data source.
+
+    Notes/Improvements:
+
+    - TBD: Should accept MSIDSet in addition to MSID as input, but this could
+      interfere with the current MSIDSet.interpolate() behavior
+    - TBD: add an option for in-place adjustment
+    - TBD: - maybe take msid as string, but manually set data source to avoid
+      'maude' 'cxc' differences...
+    - NOTE: Could use times from the msid.times span, but may create problems at
+      the beginning and end since these are subject to adjustment.  Should use
+      time query that generated the MSID
+
+    :param msid: fetch ``MSID`` or ``Msid`` object
+        MSID for which to calculate the time adjustments statistics for.
+    :param ts: str, ``CxoTime``, ``DateTime``
```

Comment:
Standard naming for generic start times is `start`, and likewise `stop`. If a time is specifically a float in CXCtime then `tstart`  / `tstop`.